File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/AlgDispenser.java
package Auxiliary_Modules;

import LivinGrimoire.Algorithm;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Random;

// (*)Algorithm Dispensers
public class AlgDispenser {
// super class to output an algorithm out of a selection of algorithms
    private ArrayList<Algorithm> algs = new ArrayList<>();
    private int activeAlg = 0;
    private Random rand = new Random();

    public AlgDispenser(Algorithm...algorithms) {
        Collections.addAll(algs, algorithms);
    }
    public AlgDispenser addAlgorithm(Algorithm alg){
        // builder pattern
        algs.add(alg);
        return this;
    }
    public Algorithm dispenseAlgorithm(){
        return algs.get(activeAlg);
    }
    public Algorithm rndAld(){
        // return a random algorithm
        return algs.get(rand.nextInt(algs.size()));
    }
    public void moodAlg(int mood){
        // set output algorithm based on number representing mood
        if ((mood>-1) && mood < (algs.size())){
            activeAlg = mood;
        }
    }
    public void algUpdate(int mood,Algorithm alg){
        // update an algorithm
        if (!((mood>-1) && mood < (algs.size()))){
            return;
        }
        algs.set(mood,alg);
    }
    public void algRemove(int mood){
        // remove an algorithm
        if (!((mood>-1) && mood < (algs.size()))){
            return;
        }
        algs.remove(mood);
    }
    public void cycleAlg(){
        activeAlg++;
        if (activeAlg == algs.size()){
            activeAlg = 0;
        }
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/AlgDispenser.kt
package auxiliary_modules

import livinGrimoire.Algorithm
import java.util.*

// (*)Algorithm Dispensers
class AlgDispenser(vararg algorithms: Algorithm) {
    // super class to output an algorithm out of a selection of algorithms
    private val algs: ArrayList<Algorithm> = ArrayList<Algorithm>()
    private var activeAlg = 0
    private val rand = Random()

    init {
        for (alg in algorithms) {
            algs.add(alg)
        }
    }

    fun addAlgorithm(alg: Algorithm): AlgDispenser {
        // builder pattern
        algs.add(alg)
        return this
    }

    fun dispenseAlgorithm(): Algorithm {
        return algs[activeAlg]
    }

    fun rndAld(): Algorithm {
        // return a random algorithm
        return algs[rand.nextInt(algs.size)]
    }

    fun moodAlg(mood: Int) {
        // set output algorithm based on number representing mood
        if (mood > -1 && mood < algs.size) {
            activeAlg = mood
        }
    }

    fun algUpdate(mood: Int, alg: Algorithm) {
        // update an algorithm
        if (!(mood > -1 && mood < algs.size)) {
            return
        }
        algs[mood] = alg
    }

    fun algRemove(mood: Int) {
        // remove an algorithm
        if (!(mood > -1 && mood < algs.size)) {
            return
        }
        algs.removeAt(mood)
    }

    fun cycleAlg() {
        activeAlg++
        if (activeAlg == algs.size) {
            activeAlg = 0
        }
    }
}